src/main/kotlin/ar/edu/unahur/obj2/vendedores/Vendedor.kt

```diff
@@ -1,51 +1,100 @@
-  fun agregarCertificacion(certificacion: Certificacion) {
-    certificaciones.add(certificacion)
-  }
+    // Si el tipo no está declarado y la función no devuelve nada, se asume Unit (es decir, vacío)
+    fun agregarCertificacion(certificacion: Certificacion) { //No retorna
+        certificaciones.add(certificacion)
+    }
 
-  fun esFirme() = this.puntajeCertificaciones() >= 30
+    fun esFirme() = this.puntajeCertificaciones() >= 30 //Tipo Bool
 
-  fun certificacionesDeProducto() = certificaciones.count { it.esDeProducto }
-  fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
+    private fun certificacionesDeProducto() = certificaciones.count { it.esDeProducto } //Tipo Int
+    fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto } //Tipo Int
 
-  fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
+    fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje } //Tipo Int
+
+    //Agregado debido al punto Vendedor Influyente, el cual nos pide como caracteristica que
+    //los vendedores puedan o no ser "influyentes", el cual cambia acorde a cada uno de ellos.
+    abstract fun esInfluyente(): Boolean //Tipo Bool
 }
 
 // En los parámetros, es obligatorio poner el tipo
-class VendedorFijo(val ciudadOrigen: Ciudad) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudad == ciudadOrigen
-  }
+class VendedorFijo(private val ciudadOrigen: Ciudad) : Vendedor() {
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {  //Tipo Bool
+        return ciudad == ciudadOrigen
+    }
+    override fun esInfluyente() = false //Tipo Bool
+
 }
 
 // A este tipo de List no se le pueden agregar elementos una vez definida
-class Viajante(val provinciasHabilitadas: List<Provincia>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return provinciasHabilitadas.contains(ciudad.provincia)
-  }
+class Viajante(private val provinciasHabilitadas: List<Provincia>) : Vendedor() {
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean { //Tipo Bool
+        return provinciasHabilitadas.contains(ciudad.provincia)
+    }
+
+    override fun esInfluyente(): Boolean { //Tipo Bool
+        return (
+                (this.provinciasHabilitadas.sumBy { provincia -> provincia.poblacion  }) >= 10000000
+                )
+    }
 }
 
-class ComercioCorresponsal(val ciudades: List<Ciudad>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudades.contains(ciudad)
-  }
+class ComercioCorresponsal(private val ciudades: List<Ciudad>) : Vendedor() { //Tipo Vendedor
+    override fun puedeTrabajarEn(ciudad: Ciudad): Boolean { //tipo bool
+        return ciudades.contains(ciudad)
+    }
+
+    override fun esInfluyente(): Boolean { //Tipo Bool
+        return ((this.ciudades.size) >= 5 || (ciudades.map { ciudad -> ciudad.provincia }).size >= 3)
```

Comment:
Dos cosas:

- Podrías delegar la segunda parte (que es medio larga) en otro método que devuelva las provincias
- La lógica de la rama derecha del OR no saca las provincias repetidas, solo está chequeando que haya más de 3 ciudades.

Es decir:
`this.ciudades` -> listado de ciudades
`this.ciudades.map{c->c.provincia}` -> listado de la provincia de cada una de esas ciudades
`this.ciudades.map{c->c.provincia}.toSet()` -> listado de provincia sin repetidos

Siguiendo el ejemplo del enunciado, un corresponsal con `ciudades=Rosario (Santa Fe), Rafaela (Santa Fe), Amstrong (Santa Fe) y Diamante (Entre Rios)` en tu código daría que es influyente aunque tenga solo 4 ciudades y 2 provincias

Reply:
ya arregle la implementacion, pase el or derecho a un val y le agregue el toSet, pero respecto al toSet...
Busque en internet y no termine de entender, ¿que es lo que hace?
Supongo que lo pasa a tipo lista (nuevamente, porque ya es una lista) pero en ese tipo de listas nuevas no se permiten repeticiones (Si me equivoco corregime, sino voy a suponer que lo que dije esta bien :P)

Reply:
Tal cual, el `toSet()` convierte la _lista_ (lista) a un _conjunto_ (set) que es un tipo de datos parecido pero que no admite repetidos